Release notes for 2020.3.3 (#106)
vircadia/vircadia-docs-sphinx: docs/source/release-notes/2020-3-3-Demeter.md

# Alpha 2020.3.3 Demeter Release Notes

> This version of Vircadia is an update to 2020.3.2. It is compatible with the 2020.3.0-Demeter protocol. This version was released on December 30th, 2020.

Reviewer: ```suggestion
This version of Vircadia is an update to 2020.3.2. It is compatible with the 2020.3.0-Demeter protocol. This version was released on January 2th, 2021.
```

    * Main "Edit" menu:
        * The two "Import Entities" actions have been brought back to the main "Edit" menu. (Since there was no unanimity about their removal.)
        * A separator has been added to properly group the 'Create' app's menu items.
    * In desktop mode only, the default window width is now 750px instead of 490px.
* App: 'Create' app material icons update. (#907)
    * Preference: "Show Zones in Create Mode" has been removed. The method behind it is still in the API, but does nothing. It has been abandoned since it was only drawing a box that wasn't useful to assist in figuring out the shape of the zone.
    * Preference: "Show Lights and Particle Systems in Create Mode" has been renamed to "Show Icons in Create Mode" since this setting was also managing all the icons (including Zones and now Parented Materials). The objective of the setting is mainly to simplify the display in edit mode when a scene is very dense.

>     * `image3D` overlays used for the icons have been migrated to Local Entities.

Reviewer 2: ```suggestion
    * `image3D` overlays used for the icons have been migrated to local entities.
```

>     * All the Entities icons (in-world non physical entities) has been updated to be more uniform. They now all have the same dark gray tone, with a white border to keep them visible under different lighting.

Reviewer 2: ```suggestion
    * All the entity icons (in-world non physical entities) have been updated to be more uniform. They now all have the same dark gray tone, with a white border to keep them visible under different lighting.
```

#### Miscellaneous

* Misc: Module updates for screenshare app. (#708, #905)
* Application: Upgrade tbb (oneAPI Threading Building Blocks) to fix a startup crash on Fedora 32 and later. (#863)
* Build: GHA now builds PR builds with crash reporting enabled on Windows. (#878)
* Application: The `--forceCrashReporting` command-line parameter has been added to enable crash reporting even if the settings have it off. (#879)
* Build: Fix hifi_qt.py reporting Ubuntu 20.04 as outdated. (#891)
* Misc: Update various references for GitHub links in the codebase. (#906)

## Domain Server

The numbers at the end of each item are the PR numbers in the Vircadia [repo](https://github.com/vircadia/vircadia).

#### General

* Server: The Domain server now sends its IP address and port to the Metaverse server with every heartbeat. (#877)

## Metaverse Server & Dashboard (Codename Iamus)

### Iamus [Metaverse Server](https://github.com/vircadia/iamus)

#### Version 2.3.6

<ul>

> <li><a href="http://github.com/vircadia/Iamus/commit/d39b5731e0481cc0006dae2e6fb4a8530137b0bb">view &bull;</a> Update ChangeLog.md</li> 

Reviewer: Why do these all have double bullet points?

Author: They are generated using a script in the Iamus repo I believe. Just a way to format it I suppose, I copy and paste it over from here https://github.com/vircadia/Iamus/blob/master/ChangeLog.md.

<li><a href="http://github.com/vircadia/Iamus/commit/c67706945d71c51d806023c02dce26d4147ea1a6">view &bull;</a> Minor bug fixes in BackupDb.sh</li> 
<li><a href="http://github.com/vircadia/Iamus/commit/6b05304c326f8303a18afbde9421ccf030949297">view &bull;</a> Update links and names (#66)</li> 
<li><a href="http://github.com/vircadia/Iamus/commit/cc8e9d64f077c2ee62405cfe2437ad3483845c24">view &bull;</a> Update NPM package version.</li> 

> <li><a href="http://github.com/vircadia/Iamus/commit/8cf1eee04aab3a069442c346241cf96aa8572f1a">view &bull;</a> Fix problem where user wasn't returned their token by /api/v1/tokens     The search field must be specified to AccountScopeFilter     A few other places were checked and updated.</li> 

Reviewer 2: ```suggestion
<li><a href="http://github.com/vircadia/Iamus/commit/8cf1eee04aab3a069442c346241cf96aa8572f1a">view &bull;</a> Fix problem where user wasn't returned their token by /api/v1/tokens. The search field must be specified to AccountScopeFilter. A few other places were checked and updated.</li> 
```

<li><a href="http://github.com/vircadia/Iamus/commit/1d0be3bf637315c3196390124600cada3fc218f1">view &bull;</a> Bump version to 2.3.6</li> 
</ul>

#### Version 2.3.5

<ul>
<li><a href="http://github.com/vircadia/Iamus/commit/e95291d4988f48d780795e134a364d9120f22fc3">view &bull;</a> Update ChangeLog.md</li> 

> <li><a href="http://github.com/vircadia/Iamus/commit/b360944a70a994ee701a0f2b23303a1d71dd76fb">view &bull;</a> Change Place filter 'maturity' to take a comma separated string     of maturity classes to match.</li> 

Reviewer 2: ```suggestion
<li><a href="http://github.com/vircadia/Iamus/commit/b360944a70a994ee701a0f2b23303a1d71dd76fb">view &bull;</a> Change Place filter 'maturity' to take a comma separated string of maturity classes to match.</li> 
```

<li><a href="http://github.com/vircadia/Iamus/commit/00355c7481b1958379d574a1ff7acc65a18d2eed">view &bull;</a> When creating a Place, default maturity to the parent domain's maturity. Default DomainEntity.maturity to Maturity.UNRATED.</li> 

> <li><a href="http://github.com/vircadia/Iamus/commit/a80864dfea02c284c195000184392f9c1f7bdcce">view &bull;</a> Massive refactoring to remove circular import dependencies that were     causing null value initializations due to how Javascript loads. Move Entity field definitions into separate files. Moved Entity field getting and setting into the Entity controlling class. Repackaged (into multiple files) Permission and Get/Set/Validate routines.</li> 

Reviewer 2: ```suggestion
<li><a href="http://github.com/vircadia/Iamus/commit/a80864dfea02c284c195000184392f9c1f7bdcce">view &bull;</a> Massive refactoring to remove circular import dependencies that were causing null value initializations due to how JavaScript loads. Move Entity field definitions into separate files. Moved Entity field getting and setting into the Entity controlling class. Repackaged (into multiple files) Permission and Get/Set/Validate routines.</li> 
```

<li><a href="http://github.com/vircadia/Iamus/commit/d45833f889f838bdefe5cf8bfb31d316aad486e5">view &bull;</a> Record network_address and network_port if specified when creating doamin.</li> 
<li><a href="http://github.com/vircadia/Iamus/commit/acaa36a494ad439122f48376ba31f08068219566">view &bull;</a> Added "maturity" and "tags" to PlaceEntity Centralized "maturity" specifications into src/Entities/sets/Maturity.ts Moved AccountAvailability and AccountRoles to src/Entities/sets and made a set pattern. Documented "/api/v1/places?maturity=LEVEL" and "/api/v1/places?tags=TAGLIST" Added comments to src/Entities/EntityFilters/CriteriaFilter.ts explaining how filters are to be used Added "maturity-catagories" to Places responses Fixed /api/v1/users/places to return only the places that belong to the requesting user Deleted bin/www -- what was that from??</li> 
<li><a href="http://github.com/vircadia/Iamus/commit/56c4649b60f58915c3b05907f3f1149cf3fe0e71">view &bull;</a> Bump version to 2.3.4</li> 
</ul>

### Iamus [Metaverse Dashboard](https://github.com/vircadia/project-iamus-dashboard)

> Updates are reflected [here](https://dashboard.vircadia.com/). The Dashboard has been upgraded from version 1.1.3 to version 1.2.2. (13 commits altering functionality, 18 commits total)

Reviewer 2: ```suggestion
Updates are reflected [here](https://dashboard.vircadia.com/). The Dashboard has been upgraded from version 1.1.3 to version 1.2.2. (13 commits altering functionality, 18 commits total.)
```
